pandas/tseries/common.py

```diff
@@ -106,6 +106,8 @@ def _delegate_property_get(self, name):
         elif not is_list_like(result):
             return result
 
+        result = np.asarray(result)
+
```

Comment:
you can change it to

``result.take(...)`` which will handle this. It was one this way *because* it was an array originally.